src/libs/actions/Report.js

```diff
@@ -323,6 +349,77 @@ function fetchChatReportsByIDs(chatList) {
         });
 }
 
+/**
+ * Updates chat and IOU data in Onyx with data from API
+ * for given chat or IOU report ID.
+ *
+ * @param {Number} reportID
+ */
+function updateIOUAndChatReportData(reportID) {
+    API.Get({
+        returnValueList: 'reportStuff',
+        reportIDList: reportID,
+        shouldLoadOptionalKeys: true,
+        includePinnedReports: true,
+    })
+        .then(({reports}) => {
+            if (reports.length === 0) {
+                return;
+            }
+            const report = Object.values(reports)[0];
+            if (lodashGet(report, 'reportNameValuePairs.type') === 'chat') {
+                // This is a chat report, find corresponding IOU report and link them
+                updateIOUReportData(report).then((iouReportObject) => {
+                    let chatReportObject = {};
+                    if (iouReportObject) {
+                        const iouReportKey = `${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportObject.reportID}`;
+                        Onyx.merge(iouReportKey, iouReportObject);
+                        chatReportObject = {
+                            iouReportID: iouReportObject.reportID,
+                            hasOutstandingIOU: iouReportObject.stateNum === 1
+                                && iouReportObject.total !== 0,
+                        };
+                    } else {
+                        chatReportObject = {
+                            iouReportID: undefined,
+                            hasOutstandingIOU: false,
+                        };
+                    }
+                    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`;
+                    Onyx.merge(reportKey, chatReportObject);
+                });
+            } else if (lodashGet(report, 'reportNameValuePairs.type') === 'iou') {
```

Comment:
This should, in my opinion, be separated into two methods each one handling the different report type.